pkg/storage/replica_proposal.go

```diff
@@ -582,6 +579,18 @@ func (r *Replica) handleReplicatedEvalResult(
 		rResult.ComputeChecksum = nil
 	}
 
+	if rResult.RaftLogDelta != nil {
+		r.mu.Lock()
+		r.mu.raftLogSize += *rResult.RaftLogDelta
+		if r.mu.raftLogSize < 0 {
+			// Ensure raftLogSize is not negative since it isn't persisted between
```

Comment:
When a node restarts the `raftLogSize` is reset to 0 even though the raft log is not empty. We then adjust this field as the raft log is truncated leading to a possibly negative value.